installer/scripts/omsadmin.sh

```diff
@@ -416,8 +422,40 @@ onboard()
 
     set_proxy_setting
 
-    # Send the request to the registration server
-    RET_CODE=`curl --header "x-ms-Date: $REQ_DATE" \
+    if [ -n "$AZURE_RESOURCE_ID" ]; then
+        if [ -f /sys/devices/virtual/dmi/id/chassis_asset_tag ]; then
+            ASSET_TAG=$(cat /sys/devices/virtual/dmi/id/chassis_asset_tag)
+            case "$ASSET_TAG" in
+                77*) OMSCLOUD_GUID=$ASSET_TAG ;;       # If the asset tag begins with a 77 this the azure guid
```

Comment:
always get the GUID and store it into the omsadmin.conf no matter the AzureResourceId is specified or not.